### ISSUE TITLE
perhaps system prompt should be session related not global

### ISSUE THREAD

**Issue by diemus** [cogentapps/chat-with-gpt#35]
great project. I was thinking build something like this by myself, just don't have time to do it. in my original plan, I think system prompt should be a session related config, thusly user can switch between different type of sessions. and there should be a prompt store. in each session, user could use different prompt templates.  just share some ideas

**Comment by mihha** [comment]
Can you please explain the purpose of that system prompt?
I have to admit that I don't understand what is that parameter used for and how can we utilize it 🤔

Thank you in advance!

**Comment by cogentapps** [comment]
Good idea. Each chat session should have its own system prompt. The system prompt should be saved with the chat, so if you go back to an old chat, it remembers the system prompt. And there can be a button below the System Prompt setting to 'make this the default' for future sessions.

I won't be able to implement this right away, but will add it to the roadmap.

**Comment by diemus** [comment]
@cogentapps I think we should also have a session based conversation compact strategy. like 

+ simple QA mode.  just answer the last question
+ scrolling mode. just put the recent N msg in the context
+ user only mode. just remember user's question and put it into the context
+ compact mode. use gpt to compact context before adding a new question.

for example, if i want set a session as a helper for translate. I could set a system prompt and a simple QA mode to save unnecessary context token consume. finish all the same type job in this session.

**Comment by crosbyh** [comment]
If possible, wonder if system prompts could be defined as templates, so users could save a library of useful system prompts as starting points for different tasks?
